### PR TITLE
Fixes #37696 - Add convert2rhel fact to host reported data facet

### PR DESCRIPTION
**Pull request by chris1984**
Convert2rhel creates a custom fact with an env variable, If it got converted through Satellite or through the Customer Portal. It looks like this:

```bash
[root@toledo8 ~]# cat /etc/rhsm/facts/convert2rhel.facts {
"conversions.version": "1",
"conversions.activity": "conversion",
"conversions.packages.0.nevra": "convert2rhel-0:2.0.0-1.el8.noarch",
"conversions.packages.0.signature": "RSA/SHA256, Thu May 30 13:31:33 2024, Key ID 199e2f91fd431d51",
"conversions.executed": "/usr/bin/convert2rhel",
"conversions.success": true,
"conversions.activity_started": "2024-07-11T17:28:54.281664Z",
"conversions.activity_ended": "2024-07-11T17:48:47.026664Z",
"conversions.source_os.id": "Cerulean Leopard",
"conversions.source_os.name": "AlmaLinux",
"conversions.source_os.version": "8.10",
"conversions.target_os.id": "Ootpa",
"conversions.target_os.name": "Red Hat Enterprise Linux",
"conversions.target_os.version": "8.10",
"conversions.env.CONVERT2RHEL_THROUGH_FOREMAN": "1", <- what we care about
"conversions.run_id": "null"
}
```

The env part of the fact is getting filtered out due to https://github.com/theforeman/foreman/blob/55d2a0d22a0fe738342804bcc4987fc24a3b0917/app/registries/foreman/settings/facts.rb#L23

We can see that the fact with this PR is present on the system when looking with pry:

```bash
[12] pry(main)> foo.reported_data
=> #<HostFacets::ReportedDataFacet:0x00007f6a30b9fb30
 id: 19,
 host_id: 18,
 boot_time: Thu, 11 Jul 2024 17:49:26.000000000 UTC +00:00,
 created_at: Mon, 29 Jul 2024 18:36:51.167876000 UTC +00:00,
 updated_at: Mon, 29 Jul 2024 19:30:23.632255000 UTC +00:00,
 virtual: true,
 sockets: 1,
 cores: 1,
 ram: 7953,
 disks_total: nil,
 kernel_version: "4.18.0-553.8.1.el8_10.x86_64",
 bios_vendor: nil,
 bios_release_date: nil,
 bios_version: nil,
 convert2rhel: 1>
```

The convert2rhel team wants that fact reported through RH cloud so they know if they should keep working on enhancing this process and to see how many people are using it.

Once this is in, I will make a PR to RH Cloud to add that into the report we send off.

### CODE REVIEW

**Comment by chris1984** [comment]
@ekohl here is the pr I made about the facts we talked about yesterday

File: app/services/katello/rhsm_fact_parser.rb [hunk 1]
```diff
@@ -117,6 +117,10 @@ def kernel_version
       facts['uname.release']
     end
 
+    def convert2rhel
+      facts['conversions.env.CONVERT2RHEL_THROUGH_FOREMAN'].to_i
```
Comment (ekohl) [inline app/services/katello/rhsm_fact_parser.rb:121]: I don't know this fact that well, but what does this actually mean? The fact itself implies it's more like a boolean.

I can see it's set here:

https://github.com/theforeman/foreman_ansible/blob/afd7faa412259f339605e5d419a039a72fb30e3b/app/views/foreman_ansible/job_templates/convert_to_rhel.erb#L30-L31

Then in convert2rhel this gets picked up:
https://github.com/oamg/convert2rhel/blob/d738c1dae0a50e0d0c5a7aad60316505fbf70e36/convert2rhel/breadcrumbs.py#L70-L71
https://github.com/oamg/convert2rhel/blob/d738c1dae0a50e0d0c5a7aad60316505fbf70e36/convert2rhel/breadcrumbs.py#L140-L150

So isn't this better?
```suggestion
    def convert2rhel_though_foreman?
      facts['conversions.env.CONVERT2RHEL_THROUGH_FOREMAN'] == "1"
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Please add a test for this in `test/unit/katello/rhsm_fact_parser_test.rb`.

**Comment by chris1984** [comment]
After talking with @ShimShtein we decided it was best to go into Katello. Closing this one in favor of the Katello one I made.

**Comment by chris1984** [comment]
https://github.com/Katello/katello/pull/11096

**Pull request closed without merging**